core/src/main/java/org/opensearch/sql/datasource/DataSourceServiceImpl.java

```diff
@@ -140,17 +173,37 @@ private Optional<DataSourceMetadata> getDataSourceMetadata(String dataSourceName
 
   private DataSource getDataSourceFromMetadata(DataSourceMetadata dataSourceMetadata) {
     if (!dataSourceMap.containsKey(dataSourceMetadata)) {
-      clearDataSource(dataSourceMetadata);
+      clearDataSourceWithSameName(dataSourceMetadata.getName());
       dataSourceMap.put(dataSourceMetadata,
           dataSourceFactoryMap.get(dataSourceMetadata.getConnector())
               .createDataSource(dataSourceMetadata));
     }
     return dataSourceMap.get(dataSourceMetadata);
   }
 
-  private void clearDataSource(DataSourceMetadata dataSourceMetadata) {
+  private void clearDataSourceWithSameName(String dataSourceName) {
     dataSourceMap.entrySet()
-        .removeIf(entry -> entry.getKey().getName().equals(dataSourceMetadata.getName()));
+        .removeIf(entry -> entry.getKey().getName().equals(dataSourceName));
   }
 
+
+  private Set<DataSourceMetadata> removeAuthInfo(Set<DataSourceMetadata> dataSourceMetadataSet) {
+    return dataSourceMetadataSet.stream()
+        .map(this::removeAuthInfo)
+        .collect(Collectors.toSet());
+  }
+
+  private DataSourceMetadata removeAuthInfo(DataSourceMetadata dataSourceMetadata) {
+    HashMap<String, String> filteredPropertiesMap = new HashMap<>();
+    dataSourceMetadata.getProperties().entrySet()
+      .stream()
+      .filter(entry -> !entry.getKey().contains("auth"))
+      .forEach(entry -> filteredPropertiesMap.put(entry.getKey(), entry.getValue()));
+    DataSourceMetadata result = new DataSourceMetadata();
+    result.setName(dataSourceMetadata.getName());
+    result.setConnector(dataSourceMetadata.getConnector());
+    result.setAllowedRoles(dataSourceMetadata.getAllowedRoles());
+    result.setProperties(filteredPropertiesMap);
+    return result;
+  }
```

Comment:
Not sure if I understand correct. I feel this maybe too complicated for easy thing?

```
DataSourceMetadata copy = dataSourceMetadata.clone(); // implement a clone method
copy.getProperties().remove("auth");
return copy;
```

Reply:
The above code will remove the properties from original object as well.
I want to make a deep copy and return objects. 
I need to rethink if I need to create a new object or return the same by removing the relevant informaiton.

Reply:
Sure, I think it depends on if we only remove a few or a lot and if this number will change very soon.